src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitor.scala

```diff
@@ -156,12 +156,11 @@ class SubmissionMonitor(workspaceName: WorkspaceName,
     }
 
   private def withWorkspaceContext[T](workspaceName: WorkspaceName, writeLock: Boolean, txn: RawlsTransaction)(op: (WorkspaceContext) => T ) = {
+    txn.withLock(workspaceName, writeLock) {
```

Comment:
maybe a bit of scope creep: since blocking is bad we could implement a tryLock and if it can't get the lock then it reposts the message it is handling to itself for future enjoyment.

To say that it can afford to block is not accurate, there is still a fixed thread pool and this blocking actor will squat on a thread until it is unblocked.

I am happy to drop a jira and leave it as is.